.github/workflows-source/ci-main.yml

```
@@ -15,9 +15,9 @@ on:
   workflow_call:
 
 # runs for the same workflow are cancelled on PRs but not on master
-# explanation: on scheduled jobs neither head_ref or ref is set, so the fallback is run_id and master will run without being cancelled
+# explanation: on push to master head_ref is not set, but ref is, so we want it to fall back to run_id so it is not cancelled
 concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.ref || github.run_id }}
+  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
```

Why have `github.ref` in there in the first place? As in, why not just:
```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

---

Ok I think I know. The `github.head_ref` is there to check if it's a PR, if so we want the group to include `github.ref`.

---

Actually I think you have a point... `head_ref` will be set if it's a PR, so it shouldn't require `ref`. Let me try that out.